Tinder.DAL/Repositories/PhotoRepository.cs

```diff
@@ -9,5 +10,58 @@ public PhotoRepository(ApplicationDbContext context) : base(context)
         {
             
         }
+
+        public override async Task<PhotoEntity> CreateAsync(PhotoEntity entity, CancellationToken cancellationToken)
+        {
+            await _dbSet.AddAsync(entity, cancellationToken);
+            await _context.SaveChangesAsync(cancellationToken);
+            return entity;
+        }
+
+        public Task<PhotoEntity?> GetByIdAsync(Guid id, Guid userId, CancellationToken cancellationToken)
+        {
+            return _dbSet.AsNoTracking()
+                .Include(p => p.User)
+                .FirstOrDefaultAsync(p => p.Id == id && p.UserId == userId, cancellationToken);
+        }
+
+        public override Task<List<PhotoEntity>> GetAllAsync(CancellationToken cancellationToken)
+        {
+            return _dbSet.AsNoTracking().Include(p => p.User).ToListAsync(cancellationToken);
+        }
+
+        public async Task<List<PhotoEntity>> UpdateRangeAsync(List<PhotoEntity> photos, CancellationToken cancellationToken)
+        {
+            foreach (var entity in photos)
+            {
+                var entityFromDb = await _dbSet.FirstOrDefaultAsync(p => p.Id == entity.Id, cancellationToken);

```

Comment:
1000 photos = 1000 db queries. It it okay?

Reply:
zalupa, soglasen